### PR TITLE
[SourceKit] Fix issue where `CompletionCheckDependencyInterval` is set to 0 when the global config request is sent

### PR DESCRIPTION
**Pull request by akyrtzi**
Fixes rdar://66309544

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci smoke test

File: tools/SourceKit/tools/sourcekitd/lib/API/Requests.cpp [hunk 1]
```diff
@@ -442,11 +442,9 @@ void handleRequestImpl(sourcekitd_object_t ReqObj, ResponseReceiver Rec) {
...
       OptimizeForIDE = EditorMode;
     }
-    Optional<unsigned> CompletionCheckDependencyInterval;
-    int64_t IntervalValue = 0;
-    if (!Req.getInt64(KeyCompletionCheckDependencyInterval,
-                      IntervalValue, /*isOptional=*/true))
-      CompletionCheckDependencyInterval = IntervalValue;
+    Optional<unsigned> CompletionCheckDependencyInterval =
+      Req.getOptionalInt64(KeyCompletionCheckDependencyInterval)
+        .map([](int64_t v)->unsigned{return v;});
```
Comment (rintaro) [inline tools/SourceKit/tools/sourcekitd/lib/API/Requests.cpp:447]: Just to clarify, should rewriting ``/*isOptional=*/true` to `/*isOptional=*/false` in the original code also fix the issue?

Comment (akyrtzi) [reply tools/SourceKit/tools/sourcekitd/lib/API/Requests.cpp:447]: Yes, but `getOptionalInt64()` is more readable. I have to admit that `getInt64()` is not well thought out as an API and rife for confusion, as in this case.

**Review by rintaro — APPROVED** [review]

File: tools/SourceKit/tools/sourcekitd/lib/API/Requests.cpp (hunk 1 above)
Comment (rintaro) [reply tools/SourceKit/tools/sourcekitd/lib/API/Requests.cpp:447]: Thanks! I agree `getOptionalInt64()` is more readable.

**Comment by akyrtzi** [comment]
@swift-ci smoke test OS X platform

**Comment by akyrtzi** [comment]
@swift-ci Please test Windows platform

**Pull request merged**